pkgs/applications/audio/feishin/default.nix

```diff
@@ -0,0 +1,103 @@
+    changelog = "https://github.com/jeffvli/feishin/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ onny ];
+  };
+
+  filename = if stdenv.isDarwin then
+    "${appname}-${version}-mac-x64.zip"
+  else
+    "${appname}-${version}-linux-x64.tar.xz";
+  src = fetchurl {
+    url =
+      "https://github.com/jeffvli/feishin/releases/download/v${version}/${filename}";
+    sha256 = if stdenv.isDarwin then
+      "sha256-WzU/Yd3cNMIpmkKWC29mQlviYXiYV5k8+80iyzilPGc="
+    else
+      "sha256-o+fEjdG2iN84FEchyAKWZCpsrqDm1K1k+Q2K+cwYBZE=";
+  };
+
+  icon = fetchurl {
+    url =
+      "https://github.com/jeffvli/feishin/raw/development/assets/icons/1024x1024.png";
+    sha256 = "sha256-8Qigt1CNMa3SDVK2cdqWJuMSl19yfy6nPQfME4qA48I=";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = "feishin";
+    desktopName = "Feishin";
+    comment = "Full-featured Subsonic/Jellyfin compatible desktop music player";
+    icon = "feishin";
+    exec = "feishin %u";
+    categories = [ "Audio" ];
+    mimeTypes = [ "x-scheme-handler/feishin" ];
+  };
+
+  linux = stdenv.mkDerivation {
+    inherit appname pname version src desktopItem icon;
+    meta = meta // { platforms = [ "x86_64-linux" "aarch64-linux" ]; };
+    nativeBuildInputs = [ makeWrapper graphicsmagick ];
+
+    # Installs mpv as a requirement
+    propagatedBuildInputs = [ mpv ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      mkdir -p $out/share
+      cp -r resources $out/share/${pname}/
+      cp -r locales $out/share/${pname}/
+
+      makeWrapper ${electron_24}/bin/electron $out/bin/${pname} \
+        --add-flags $out/share/${pname}/app.asar
+      install -m 444 -D "${desktopItem}/share/applications/"* \
+        -t $out/share/applications/
+      for size in 16 24 32 48 64 128 256 512; do
+        mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+        gm convert -resize "$size"x"$size" ${icon} $out/share/icons/hicolor/"$size"x"$size"/apps/${appname}.png
+      done
+      runHook postInstall
+    '';
+
+    shellHook = ''
+      set -x
+      export LD_LIBRARY_PATH=${mpv}/lib
+      set +x
+    '';
+  };
+
+  darwin = stdenv.mkDerivation {
+    inherit pname version src appname;
+    meta = meta // { platforms = [ "x86_64-darwin" "aarch64-darwin" ]; };
+    sourceRoot = "${appname}.app";
+
+    nativeBuildInputs = [ makeWrapper unzip ];
+
+    # Installs mpv as a requirement
+    propagatedBuildInputs = [ mpv ];
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out/{Applications/${appname}.app,bin}
+      cp -R . $out/Applications/${appname}.app
+      makeWrapper $out/Applications/${appname}.app/Contents/MacOS/${appname} $out/bin/${pname}
+      runHook postInstall
+    '';
+
+    shellHook = ''
+      set -x
+      export LD_LIBRARY_PATH=${mpv}/lib
+      set +x
+    '';
+  };
+in if stdenv.isDarwin then darwin else linux
```

Comment:
To be honest, this expression looks unnecessarily complicated. I vote for
- Split it into two files, one for each system
- `default.nix` would call the respective file